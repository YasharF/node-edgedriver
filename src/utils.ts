import os from 'node:os'
import fs from 'node:fs/promises'

export function skipDownload () {
  return (
    process.env.npm_config_edgedriver_skip_download ||
    process.env.EDGEDRIVER_SKIP_DOWNLOAD
  )
}

export async function hasAccess (filePath: string) {
  return fs.access(filePath).then(() => true, () => false)
}

export function findByArchitecture (name: string) {
  const arch = ['arm64', 'ppc64', 'x64', 's390x'].includes(os.arch())
    ? '64' + (os.platform() === 'darwin' ? '_m1' : '')
    : '32'
  const platformIdentifier = os.platform() === 'win32'
    ? 'win'
    : os.platform() === 'darwin'
      ? 'mac'
      : 'arm' // linux
  return name.split('.')[0].toLocaleLowerCase().endsWith(`_${platformIdentifier}${arch}`)
}
