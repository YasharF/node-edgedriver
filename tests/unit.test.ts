import os from 'node:os'
import { vi, test, expect } from 'vitest'

import { findByArchitecture, parseParams } from '../src/utils.js'

vi.mock('node:os', () => ({
  default: {
    arch: vi.fn(),
    platform: vi.fn()
  }
}))

test('findByArchitecture', () => {
  vi.mocked(os.arch).mockReturnValue('arm')
  vi.mocked(os.platform).mockReturnValue('linux')
  expect(findByArchitecture('edgedriver_linux32.zip')).toBe(true)
  vi.mocked(os.arch).mockReturnValue('arm64')
  vi.mocked(os.platform).mockReturnValue('linux')
  expect(findByArchitecture('edgedriver_linux64.zip')).toBe(true)
  vi.mocked(os.arch).mockReturnValue('arm')
  vi.mocked(os.platform).mockReturnValue('win32')
  expect(findByArchitecture('edgedriver_win32.zip')).toBe(true)
  vi.mocked(os.arch).mockReturnValue('arm64')
  vi.mocked(os.platform).mockReturnValue('win32')
  expect(findByArchitecture('edgedriver_win64.zip')).toBe(true)
  vi.mocked(os.arch).mockReturnValue('x64')
  vi.mocked(os.platform).mockReturnValue('darwin')
  expect(findByArchitecture('edgedriver_mac64.zip')).toBe(true)
  vi.mocked(os.arch).mockReturnValue('arm64')
  vi.mocked(os.platform).mockReturnValue('darwin')
  expect(findByArchitecture('edgedriver_mac64_m1.zip')).toBe(true)
})

test('parseParams', () => {
  expect(parseParams({ baseUrl: 'foobar', silent: true, verbose: false, allowedIps: ['123', '321'] }))
    .toMatchSnapshot()
})
