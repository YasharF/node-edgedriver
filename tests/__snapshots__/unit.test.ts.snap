// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parseParams 1`] = `
[
  "--base-url=foobar",
  "--silent",
  "--allowed-ips=123",
  "--allowed-ips=321",
]
`;
